### PR TITLE
Create printHelp: the help helper

### PR DESCRIPTION
**Pull request by jaynagpaul**
#28 

Would love any feedback/requested changes, no matter how nitpicky.

### CODE REVIEW

**Comment by MasterQ32** [comment]
Can you add a usage example and tests for help printing?

**Comment by jaynagpaul** [comment]
Yes, I have added a usage of demo.zig, and two tests. 

The example given in the original issue should work as expected with the exception that I chose to explicitly set the executable name in meta to simplify the interface (avoid allocations / an extra argument for passing in the name):
```zig
const CliOptions = struct {
  help: bool = true,
  output: ?[]const u8 = null,
  long: u32 = 0,
  pub const shorthands = .{
    .h = "help",
    .o = "output",
  };
  pub const meta = .{
    .summary = "[-h] [-o] [--long] <files>",
    .full_text = "This command prints the current time.",
    .option_docs = .{
      .name = "cli.zig",
      .help = "Prints this help.",
      .output = "Sets the output file.",
      .long = "Sets the number of lines."
    },
  }
};

// ...

if(cli.options.help) {
  try args_parser.printHelp(CliOptions, std.io.getStdout().writer());
  return 0;
}
```

**Comment by MasterQ32** [comment]
> (avoid allocations / an extra argument for passing in the name):

Please add this from the original argument parser. Hardcoding as a bad idea because people will see this:

```sh-session
[felix@denkplatte-v2 compi-index]$ dog --help
Usage: cat [OPTION]... [FILE]...
Concatenate FILE(s) to standard output.
…
```

Remove the option from the spec and make the exe name non-option (so the user is forced to pass one in), no allocation required, then i'm happy :)

**Comment by jaynagpaul** [comment]
Fixed!

File: demo.zig [hunk 1]
```diff
@@ -58,6 +57,6 @@ pub fn main() !u8 {
         std.debug.print("\t'{s}'\n", .{arg});
     }
 
-    try argsParser.printHelp(Options, std.io.getStdOut().writer());
+    try argsParser.printHelp(Options, "demo.zig", std.io.getStdOut().writer());
```
Comment (MasterQ32) [inline demo.zig:60]:
```diff
- try argsParser.printHelp(Options, "demo.zig", std.io.getStdOut().writer());
+ try argsParser.printHelp(Options, options.executable_name orelse "demo", std.io.getStdOut().writer());
```

**Review by MasterQ32 — CHANGES_REQUESTED** [review]
A teeeny tiny nitpick still :grin: 

You usually don't use zig files on the command line, but executables. in that case the exe is either `demo.exe` or `demo` depending on the os, not `demo.zig` :)

File: demo.zig (hunk 1 above)
Comment (jaynagpaul) [reply demo.zig]: Changed! 

Just out of curiosity, options.executable_name currently includes the full path to the executable. Would it be better to rename executable_name to executable_path? Or truncate to just the stem of the executable?

Comment (MasterQ32) [reply demo.zig]: > Just out of curiosity, options.executable_name currently includes the full path to the executable

This actually fully depends on the OS! Check this out:
```sh-session
[felix@denkplatte-v2 zig-args]$ ./demo
executable name: ./demo
[felix@denkplatte-v2 zig-args]$ ../zig-args/demo
executable name: ../zig-args/demo
[felix@denkplatte-v2 zig-args]$ $(realpath ../zig-args/demo )
executable name: /home/felix/projects/libraries/zig-args/demo
[felix@denkplatte-v2 zig-args]$ export "PATH=$(pwd):$PATH"
[felix@denkplatte-v2 zig-args]$ demo
executable name: demo
[felix@denkplatte-v2 zig-args]$ ln -s demo foobar
[felix@denkplatte-v2 zig-args]$ ../zig-args/foobar
executable name: ../zig-args/foobar
```

How the user processes and displays this information is up to them

**Review by MasterQ32 — APPROVED** [review]

**Comment by MasterQ32** [comment]
Thanks!

File: demo.zig (hunk 1 above)
Comment (jaynagpaul) [reply demo.zig]: Interesting, thanks!

**Pull request merged**